sdks/python/apache_beam/runners/common.py

```diff
@@ -512,7 +514,7 @@ def _reraise_augmented(self, exn):
           traceback.format_exception_only(type(exn), exn)[-1].strip()
           + step_annotation)
       new_exn._tagged_with_step = True
-    raise new_exn, None, original_traceback
+    six.raise_from(new_exn, original_traceback)
```

Comment:
This use of `raise_from` instead of `reraise` led to a bug with dropped stacktraces: https://issues.apache.org/jira/projects/BEAM/issues/BEAM-3956

Are the other uses of `raise_from` instead of `reraise` in this PR appropriate?